lib/elementWrapper/dollarWrapper.js

```diff
@@ -17,12 +20,31 @@ class DollarWrapper extends ElementWrapper {
   constructor(attrValuePairs, _options, ...args) {
     super('CustomSelector', 'query', attrValuePairs, _options, ...args);
     this._get = async () => {
-      return await handleRelativeSearch(
-        await (this.selector.label.startsWith('//') || this.selector.label.startsWith('(')
-          ? $$xpath(this.selector.label, this._options.selectHiddenElements)
-          : $$(this.selector.label, this._options.selectHiddenElements)),
-        this.selector.args,
-      );
+      function functionSelector(argument) {
+        let { querySelector, args } = argument;
+        eval(`querySelector = ${querySelector}`);
+        const queryResult = querySelector(args);
```

Comment:
I think the function should not just be limited to use of the query selector for example I should be able to 

```
$((id) => document.getElementById(id), { args: '#foo' })
```

Is that covered too?

Reply:
yes they are covered too..this is just the name given to user's predicate function which will be executed in browsers context

Reply:
Great, can you add some examples on the jsdoc to list that?